src/Contacts/include/BipedalLocomotion/ContactDetectors/ContactDetector.h

```diff
@@ -8,105 +8,102 @@
-    * @return container of contacts
-    */
-    virtual const EstimatedContactList& getOutput() const final;
+     * Reset the internal contact states of the detector to false
+     * @return True in case of success, false otherwise.
+     */
+    bool resetContacts();
 
     /**
      * Get state of specific contact
      * @param[in] contactName name of contact
      * @param[out] contact estimated contact
      * @return true if contact exists, false otherwise
      */
-    bool get(const std::string& contactName, BipedalLocomotion::Contacts::EstimatedContact& contact) const;
+    bool get(const std::string& contactName, Contacts::EstimatedContact& contact) const;
 
     /**
      * Get state of specific contact
      * @param[in] contactName name of contact
      * @return contact state if contact exists, a dummy contact otherwise
      */
-    BipedalLocomotion::Contacts::EstimatedContact get(const std::string& contactName) const;
-
-    /**
-    * Determines the validity of the object retrieved with get()
-    * @return True in case of success, false otherwise.
-    */
-    virtual bool isOutputValid() const final;
+    Contacts::EstimatedContact get(const std::string& contactName) const;
 
 protected:
     /**
-    * These custom parameter specifications should be specified by the derived class.
-    * @param[in] handler configure the custom parameters for the estimator
-    * @return True if success, false otherwise
-    */
-    virtual bool customInitialization(std::weak_ptr<const BipedalLocomotion::ParametersHandler::IParametersHandler> handler);
+     * These custom parameter specifications should be specified by the derived class.
+     * @param[in] handler configure the custom parameters for the estimator
+     * @return True if success, false otherwise
+     */
+    virtual bool
+    customInitialization(std::weak_ptr<const ParametersHandler::IParametersHandler> handler);
 
     /**
-    * The derived class must implement the contact detection technique to update the contact states
-    * @return True if success, false otherwise
-    */
-    virtual bool updateContactStates();
-
+     * The derived class must implement the contact detection technique to update the contact states
+     * @return True if success, false otherwise
+     */
+    virtual bool updateContactStates() = 0;
 
     /**
-    * Enumerator used to determine the running state of the estimator
-    */
+     * Enumerator used to determine the running state of the estimator
+     */
     enum class State
     {
-        NotInitialized, /**< The estimator is not initialized yet call FloatingBaseEstimator::initialze
-                           method to initialize it*/
-        Initialized,    /**< The estimator is initialized and ready to be used */
-        Running         /**< The estimator is running */
+        NotInitialized, /**< The estimator is not initialized yet call
+                           FloatingBaseEstimator::initialze method to initialize it*/
```

Comment:
Leftover from previous PRs 
```suggestion
                           ContactDetector::initialize method to initialize it*/
```